### PR TITLE
remove redundant BA damage recursion

### LINKED ISSUE #2620: Arrow VI Homing Artillery vs hex containing Elemental Battle Armor
### Environment
47.16 MacOS

### Description
Bug- Arrow IV Homing Artillery against hex containing Elemental Battle Armor [AOE Attack].
A Sprint VTOL uses TAG to tag the hex some Elementals are in. (The hex, not the BA)
An Arrow IV round lands in that hex.
By my understanding of the rules in TacOps it should do 5 damage to each Elemental.
For some reason in 47.16 it seems to be doing 3 lot of 5 damage.
TacOps AR pg 150
TacOps AUE pg 166

### Files
Game Log attached
Screen Shot 2021-01-31 at 23 26 47

### PR DESCRIPTION
**Pull request by NickAragua**
Fixes #2620 by removing a redundant loop through all BA suits from the artillery weapon indirect homing handler - it's already done in the damageEntity method.

### CODE REVIEW

**Comment by NickAragua** [comment]
Also, per errata, the hex takes 20 instead of 10 damage.

File: megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectHomingHandler.java [hunk 1]
```diff
@@ -287,7 +287,7 @@ public boolean handle(IGame.Phase phase, Vector<Report> vPhaseReport) {
         bldgAbsorbs = Math.min(bldgAbsorbs, ratedDamage);
         // assumption: homing artillery splash damage is area effect.
         // do damage to woods, 2 * normal damage (TW page 112)
```
Comment (neoancient) [inline megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectHomingHandler.java:289]: This comment should be changed to fit the implementation.

**Review by neoancient — APPROVED** [review]
I know I fixed this same issue before. It was probably in the non-homing handler and I didn't realize there was another handler that could have the same issue.

**Comment by NickAragua** [comment]
Made a slight logic adjustment - per TacOps and @HammerGS 's post, if the hex itself is targeted for clearing, then it should take the full damage.

**Review by neoancient — APPROVED** [review]

**Pull request merged**